packages/grid/x-data-grid-pro/src/hooks/features/detailPanel/useGridDetailPanel.ts

```diff
@@ -74,10 +74,19 @@ export const useGridDetailPanel = (
 
   const handleCellKeyDown = React.useCallback<GridEventListener<GridEvents.cellKeyDown>>(
     (params, event) => {
-      if (!event.ctrlKey || event.key !== 'Enter' || props.getDetailPanelContent == null) {
-        return;
+      if (
+        (event.ctrlKey || event.metaKey) &&
+        event.key === 'Enter' &&
+        props.getDetailPanelContent != null
+      ) {
+        apiRef.current.toggleDetailPanel(params.id);
+      }
+
+      if (params.field === GRID_DETAIL_PANEL_TOGGLE_FIELD && event.key === ' ') {
```

Comment:
Instead of early return, I've split the logic info two branches handling `Ctrl+Enter` and `Space` separately.